src/guide/advanced/wasm.md

```diff
@@ -16,36 +16,41 @@ the blocks, but is also easy to understand. Something that looks like your
 traditional `if` statements and `for` loops.
 
 However, in the interim, we have decided to use another portable binary
-standard called **Web assembly** or **WASM**. The main advantage of this
-format is that you can use any language you like (as long as it links
-statically against our helper library), and produce a 32-bit portable
-executable. The compilers take care of the optimisation, and you don't have
-to learn a new language (ahem… solidity… ahem), just to operate on the
-blockchain.
+standard called **Web assembly**, or **WASM**.
+
+## Working with WASM
+
+The main advantage of using the WASM format is that you can use any
+language you like (as long as it links statically against our helper
+library), and produce a 32-bit portable executable. The compilers take care
+of the optimisation, and you don't have to learn a new language (ahem...
+solidity... ahem), just to operate on the blockchain.
 
 You'd still need to use ISI from inside your WASM binary to do anything
-on-chain, but you can do other things, and write e.g. Rust instead of
+on-chain, but you can do other things, and write in Rust instead of
```

Comment:
```suggestion
on-chain, as we explained earlier. While the language of ISI is turing-complete (and in theory that means that you can do anything you want), it's less convenient (and efficient) to use metadata as memory, and write complex conditionals using just the tools that we've provided in the `Expression` and ISI infrastructure. Simply put it's easier to write in a well-known programming language (Rust) and build the logic out of simple instructions, rather than trying to reinvent the wheel using ISI. 
```